app/src/main/java/com/keylesspalace/tusky/MainActivity.kt

```diff
@@ -0,0 +1,606 @@
+        startActivity(composeIntent)
+        finish()
+    }
+
+    private fun setupDrawer(savedInstanceState: Bundle?) {
+
+        drawerToggle = ActionBarDrawerToggle(this, mainDrawerLayout, mainToolbar, com.mikepenz.materialdrawer.R.string.material_drawer_open, com.mikepenz.materialdrawer.R.string.material_drawer_close)
+
+        header = AccountHeaderView(this).apply {
+            headerBackgroundScaleType = ImageView.ScaleType.CENTER_CROP
+            currentHiddenInList = true
+            onAccountHeaderListener = { _: View?, profile: IProfile, current: Boolean -> handleProfileClick(profile, current) }
+            addProfile(ProfileSettingDrawerItem().apply {
+                identifier = DRAWER_ITEM_ADD_ACCOUNT
+                nameRes = R.string.add_account_name
+                descriptionRes = R.string.add_account_description
+                iconicsIcon = GoogleMaterial.Icon.gmd_add
+            }, 0)
+            attachToSliderView(mainDrawer)
+            dividerBelowHeader = false
+            closeDrawerOnProfileListClick = true
+        }
+
+        header.accountHeaderBackground.setColorFilter(ContextCompat.getColor(this, R.color.header_background_filter))
+        header.accountHeaderBackground.setBackgroundColor(ContextCompat.getColor(this, R.color.tusky_grey_10))
+        val animateAvatars = PreferenceManager.getDefaultSharedPreferences(this)
+                .getBoolean("animateGifAvatars", false)
+
+        DrawerImageLoader.init(object : AbstractDrawerImageLoader() {
+            override fun set(imageView: ImageView, uri: Uri, placeholder: Drawable, tag: String?) {
+                if (animateAvatars) {
+                    Glide.with(imageView.context)
+                            .load(uri)
+                            .placeholder(placeholder)
+                            .into(imageView)
+                } else {
+                    Glide.with(imageView.context)
+                            .asBitmap()
+                            .load(uri)
+                            .placeholder(placeholder)
+                            .into(imageView)
+                }
+            }
+
+            override fun cancel(imageView: ImageView) {
+                Glide.with(imageView.context).clear(imageView)
+            }
+
+            override fun placeholder(ctx: Context, tag: String?): Drawable {
+                if (tag == DrawerImageLoader.Tags.PROFILE.name || tag == DrawerImageLoader.Tags.PROFILE_DRAWER_ITEM.name) {
+                    return ctx.getDrawable(R.drawable.avatar_default)!!
+                }
+
+                return super.placeholder(ctx, tag)
+            }
+        })
+
+        mainDrawer.apply {
+            addItems(
+                    PrimaryDrawerItem().apply { identifier = DRAWER_ITEM_EDIT_PROFILE; nameRes = R.string.action_edit_profile; isSelectable = false; iconicsIcon = GoogleMaterial.Icon.gmd_person },
```

Comment:
I'm not a fan of it being oneline. I get that it will get long my head hurts from looking at this

Reply:
And if we defined click listener on it, we wouldn't even have to write another long when below